File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/DebugUtils.java
package me.ycdev.android.lib.common.utils;

import android.annotation.TargetApi;
import android.os.Build;
import android.os.StrictMode;

public class DebugUtils {
    /**
     * Should only be invoked in debug version. Never invoke this method in release version!
     */
    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    public static void enableStrictMode() {
        // thread policy
        StrictMode.ThreadPolicy.Builder threadPolicyBuilder =
                new StrictMode.ThreadPolicy.Builder()
                        .detectAll()
                        .penaltyLog();
        threadPolicyBuilder.penaltyFlashScreen();
        threadPolicyBuilder.penaltyDeathOnNetwork();
        StrictMode.setThreadPolicy(threadPolicyBuilder.build());

        // VM policy
        StrictMode.VmPolicy.Builder vmPolicyBuilder =
                new StrictMode.VmPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .penaltyDeath();
        StrictMode.setVmPolicy(vmPolicyBuilder.build());
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/DebugUtils.kt
package me.ycdev.android.lib.common.utils

import android.os.StrictMode

object DebugUtils {
    /**
     * Should only be invoked in debug version. Never invoke this method in release version!
     */
    fun enableStrictMode() {
        // thread policy
        val threadPolicyBuilder = StrictMode.ThreadPolicy.Builder()
            .detectAll()
            .penaltyLog()
        threadPolicyBuilder.penaltyFlashScreen()
        threadPolicyBuilder.penaltyDeathOnNetwork()
        StrictMode.setThreadPolicy(threadPolicyBuilder.build())

        // VM policy
        val vmPolicyBuilder = StrictMode.VmPolicy.Builder()
            .detectAll()
            .penaltyLog()
            .penaltyDeath()
        StrictMode.setVmPolicy(vmPolicyBuilder.build())
    }
}
